src/components/FormElements/AssetPicker.tsx

```diff
@@ -0,0 +1,311 @@
+    {
+      id: "credit_alphanum4",
+      label: "Alphanumeric 4",
+      value: {
+        type: "credit_alphanum4",
+        code: "",
+        issuer: "",
+      },
+    },
+    {
+      id: "credit_alphanum12",
+      label: "Alphanumeric 12",
+      value: {
+        type: "credit_alphanum12",
+        code: "",
+        issuer: "",
+      },
+    },
+    // TODO: add Liquidity Pool shares (for Change Trust operation)
+  ];
+
+  if (includeNative) {
+    objectOptions = [
+      {
+        id: "native",
+        label: "Native",
+        value: {
+          type: "native",
+          code: "",
+          issuer: "",
+        },
+      },
+      ...objectOptions,
+    ];
+  }
+
+  // Extra helper function to make TypeScript happy to get the right type
+  const handleOnChange = (
+    id: AssetType | undefined,
+    value: string | AssetObjectValue | undefined,
+  ) => {
+    if (!value) {
+      onChange(id, undefined);
+    }
+
+    if (variant === "string") {
+      onChange(id, value as string);
+    } else {
+      onChange(id, value as AssetObjectValue);
+    }
+  };
+
+  const handleOptionChange = (
+    optionId: AssetType | undefined,
+    optionValue: string | AssetObjectValue | undefined,
+  ) => {
+    handleOnChange(optionId, optionValue);
+    setStateValue({ type: optionId, code: "", issuer: "" });
+    setError(initErrorState);
+  };
+
+  const validateCode = (code: string, assetType: AssetType | undefined) => {
+    if (!code) {
+      return "Asset code is required.";
+    }
+
+    let minLength;
+    let maxLength;
+
+    switch (assetType) {
+      case "credit_alphanum4":
+        minLength = 1;
+        maxLength = 4;
+        break;
+      case "credit_alphanum12":
+        minLength = 5;
+        maxLength = 12;
+        break;
+      default:
+        minLength = 1;
+        maxLength = 12;
+    }
+
+    if (!code.match(/^[a-zA-Z0-9]+$/g)) {
+      return "Asset code must consist of only letters and numbers.";
+    } else if (code.length < minLength || code.length > maxLength) {
+      return `Asset code must be between ${minLength} and ${maxLength} characters long.`;
+    }
+
+    return undefined;
+  };
+
+  const handleCodeError = (value: string) => {
+    const codeError = validateCode(value, stateValue.type);
+    setError({ ...error, code: codeError || "" });
+    return codeError;
+  };
+
+  return (
+    <div className="RadioPicker__inset">
+      <RadioPicker
+        id={id}
+        selectedOption={selectedOption}
+        label={label}
+        labelSuffix={labelSuffix}
+        onChange={handleOptionChange}
+        options={variant === "string" ? stringOptions : objectOptions}
+        fitContent={fitContent}
+      />
+
+      <ExpandBox
+        isExpanded={Boolean(
+          selectedOption &&
+            ["issued", "credit_alphanum4", "credit_alphanum12"].includes(
+              selectedOption,
+            ),
+        )}
+      >
+        <AssetPickerFields
+          id={id}
+          code={{
```

Comment:
note to myself to will follow this style!

Reply:
We might need to remove `onBlur` piece and always validate on `onChange`. :sweat_smile: The current PR I'm working on will slightly change the form element API.